File: search/depth_first_search/depth_first_search.py
from collections import deque
# Node to build our tree
class Node():
    def __init__(self, data=None, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right


def depth_first_search(root, path):
    if root:
        path.append(root.data)
        depth_first_search(root.left, path)
        depth_first_search(root.right, path)

    return path

File: algorithms/search/depth_first_search/depth_first_search.rb


class DFS
  attr_accessor :root, :found_node

  def initialize(root:)
    @root = root
    @found_node = nil
  end

  def depth_first_search(root: self.root, target:)
    if root

      self.found_node = root if root.data == target

      depth_first_search(root: root.left, target: target)
      depth_first_search(root: root.right, target: target)
    end
  end

end
